### PR TITLE
Allow running build scripts from anywhere

### PR DESCRIPTION
**Pull request by jimmc**
This also allows running just sources/web/build.sh to compile the typescript files, which I will be using in a later PR to implement live mode.

### CODE REVIEW

File: containers/base/build.sh [hunk 1]
```diff
@@ -20,16 +20,18 @@
 
 # Build the docker image
 if [ -n "$1" ]; then
-  src_pydatalab="$1"
+  src_pydatalab=$(realpath "$1")
 
   # Append trailing / if needed.
   if [ "${src_pydatalab: -1}" != "/" ]; then
```
Comment (yebrahim) [inline containers/base/build.sh]: We don't need this if statement in this case, we can just append the slash.

Comment (yebrahim) [inline containers/base/build.sh]: A better fix is of course to make the code accept the path without the slash, but for the purposes of this PR, removing the if check is enough.

File: containers/datalab/build.sh [hunk 2]
```diff
@@ -25,13 +25,21 @@ VERSION_SUBSTITUTION="s/_version_/0.5.$VERSION/"
 COMMIT=`git log --pretty=format:'%H' -n 1`
 COMMIT_SUBSTITUTION="s/_commit_/$COMMIT/"
 
+if [ -z "$1" ]; then
```
Comment (yebrahim) [inline containers/datalab/build.sh]: Why do we need this change? `pydatalabPath` isn't used anywhere in this script, other than being passed without modification to `base/build.sh`, which takes care of getting its `realpath`.

File: containers/base/build.sh (hunk 1 above)
Comment (jimmc) [inline containers/base/build.sh]: The trailing slash is required by the rsync command, which behaves differently if it is not there. I moved adding the trailing slash to the rsync line.

File: containers/datalab/build.sh (hunk 2 above)
Comment (jimmc) [inline containers/datalab/build.sh]: realpath (called in the else clause) fails if you pass it an empty string.

File: containers/base/build.sh (hunk 1 above)
Comment (yebrahim) [inline containers/base/build.sh]: Right, thanks for fixing that.

**Pull request merged**